xbmc/view/GUIViewState.cpp

```diff
@@ -101,7 +101,15 @@ CGUIViewState* CGUIViewState::GetViewState(int windowId, const CFileItemList& it
     return new CGUIViewStateLibrary(items);
 
   if (items.IsPlayList())
-    return new CGUIViewStateMusicPlaylist(items);
+  {
+    // Playlists (like .strm) can be music or video type
+    // we set the playlist type based on the current
+    // GUI NAV window, otherwise fallback to general
+    if (windowId == WINDOW_VIDEO_NAV)
```

Comment:
maybe we need a sane default here? What if the window ID is different from both?

Reply:
add the "else" case